packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+        connectedContainers.push(mainContainer);
+        await provider.ensureSynchronized();
+        const summarizeResults = mainContainerDataStore.containerRuntime.summarizeOnDemand({
+            reason: "GC Testing",
+        });
+
+        const summarySubmitted = await summarizeResults.summarySubmitted;
+        assert(summarySubmitted.success === true, "Expected successful summarizeOnDemand!");
+        assert(summarySubmitted.data !== undefined, "Expected summarizeOnDemand data!");
+        assert(summarySubmitted.data.stage === "submit", "Should have been submitted");
+        assert(summarySubmitted.data.summaryTree !== undefined, "summary tree should exist");
+        const summaryTree = summarySubmitted.data.summaryTree;
+        uploadedSummaries.push(summaryTree);
+
+        await sleep(defaultSessionExpiryDurationMs);
+
+        await loadContainer();
+        assert(mainContainer.closed === true, "Expected mainContainer to be closed!");
+        assert(uploadedSummaries.length > 0, `${uploadedSummaries.length}`);
+    });
+
+    const randomInteger = (min: number, max: number): number => {
+        return makeRandom().integer(min, max);
+    };
+
+    const loadNewContainer = async () => {
+        const newContainer = await loadContainer();
+        newContainer.on("closed", () => {
+            const index = connectedContainers.indexOf(newContainer);
+            assert(index >= 0, "Expected container to have been added to connectedContainers");
+            closedContainers.push(connectedContainers[index]);
+            connectedContainers.splice(index, 1);
+        });
+        connectedContainers.push(newContainer);
+    };
+
+    const closeRandomContainer = async () => {
+        if (connectedContainers.length <= 0) {
+            return;
+        }
+        const index = randomInteger(0, connectedContainers.length - 1);
+        return new Promise((resolve) => resolve(connectedContainers[index].close()));
+    };
+
+    const createDataStoreForContainer = async (container: IContainer, dataStoreType: string = "TestDataObject") => {
+        const defaultDataStore = await requestFluidObject<TestDataObject>(container, "default");
+        const newDataStore = await defaultDataStore.containerRuntime.createDataStore(dataStoreType);
+        const newDataObject = await requestFluidObject<TestDataObject>(newDataStore, "");
+        testNodes.set(newDataObject.id, new TestNode(newDataObject.id));
+        unrefDataStores.add(newDataObject.id);
+        return newDataObject;
+    };
+
+    const createDataStoreForRandomContainer = async () => {
+        if (connectedContainers.length <= 0) {
+            return;
+        }
+        await createDataStoreForContainer(getRandomContainer());
+    };
+
+    const getTestNode = (id: string): TestNode => {
+        const testNode = testNodes.get(id);
+        assert(testNode !== undefined, `Datastore ${id} should exist in datastores!`);
```

Comment:
nit: How about this for the message: `DataStore with ID [${id}] not found in testNodes collection`